### ISSUE TITLE
BUG [TS2416]: Incorrect error reporting

### ISSUE THREAD

**Issue by dalisoft** [microsoft/TypeScript#42858]
# Bug Report

### 🔎 Search Terms

- [x] ts2416 class can't extend interface
- [x] typescript interface string type cant assign
- [x] Full message copy of error

### 🕗 Version & Regression Information

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzhARgZwKZQG4EMUA2GcAvnAGZQQhwDkIqAHrQNwBQbGjkscwAdjCzlcAY2IBJAMoxoGAIJgwiNnDgB3AQBMI6gOrAtMABYAuOPwCuIFFnZrN-HeoASGYAHNjMc1Zt3VODBcDwVRGGBsDHMUCAgiXH57PjQAGQhcLQEPGLiEpMCCRI9LEOi6DH5aOAAfOihLarraGBhWNhI2USK0NDhFZVAwIhBKmD7pWSgFJRU1AHp5uABRfmxgKn5RwWR0LDxCDDRAgAFUTBx8Ig1tXQMjYzgAXjgABnZT8-2r4kdnN083mebw+ajOe0uhyCZXk4UixBeMAaGFBcEWcAACnp5HBKutNttxp8IQdrsA0hksvwPMCkZYURwFksAKoXXYXUlHYkcn5wIrU0qhYG0SrtTqcbjQeBaDAiSwEeD8DDqfpKAAUAEoWEA)

### 💻 Code

```ts
export interface IStoreApp {
  windowWidth: number;
  windowHeight: number;
  pageActive: boolean;
  isLoading: boolean;
  language: 'en' | 'ru' | 'tt'; // <-- This line
}
class App implements IStoreApp {
  // Environment observables
  @observable windowWidth = 0;

  @observable windowHeight = 0;

  @observable pageActive = true;

  // PWA environments
  @observable isLoading = true;

  // User observables
  @observable language = 'en'; // <-- This line
}
export default new App();
```

### 🙁 Actual behavior

```
Property 'language' in type 'App' is not assignable to the same property in base type 'IStoreApp'.
  Type 'string' is not assignable to type '"en" | "ru" | "tt"'.
```

### 🙂 Expected behavior

Working type validating with one of three available language

**Comment by MartinJohns** [comment]
Duplicate of #10570.
